File: src/test/java/de/pflugradts/passbird/application/util/PathFaker.java
package de.pflugradts.passbird.application.util;

import lombok.AccessLevel;
import lombok.RequiredArgsConstructor;

import java.io.File;
import java.nio.file.Path;

import static org.mockito.BDDMockito.given;
import static org.mockito.Mockito.mock;

@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public class PathFaker {

    private Path path;

    public static PathFaker faker() {
        return new PathFaker();
    }

    public PathFaker fakePath() {
        path = mock(Path.class);
        return this;
    }

    public PathFaker withFileResolvingToFilename(final File file, final String filename) {
        final var resolvedPath = mock(Path.class);
        given(resolvedPath.toFile()).willReturn(file);
        given(path.resolve(filename)).willReturn(resolvedPath);
        return this;
    }

    public PathFaker withFileRepresentation(final File fileRepresentation) {
        given(path.toFile()).willReturn(fileRepresentation);
        return this;
    }

    public Path fake() {
        return path;
    }

}

File: src/test/kotlin/de/pflugradts/passbird/application/util/PathFaker.kt
package de.pflugradts.passbird.application.util

import io.mockk.every
import io.mockk.mockk
import java.nio.file.Path

fun fakePath(
    instance: Path = mockk(relaxed = true),
    resolvingTo: Pair<Path, String>? = null,
    exists: Boolean = false,
    isDirectory: Boolean = false,
    withParent: Path? = null,
): Path {
    every { instance.toFile().exists() } returns exists
    every { instance.toFile().isDirectory() } returns isDirectory
    resolvingTo?.let { every { instance.resolve(it.second) } returns it.first }
    every { instance.toFile().parentFile } returns withParent?.toFile()
    return instance
}
